pact/provider/core/src/test/java/org/arquillian/algeron/pact/provider/core/PactsReaderTest.java

```diff
@@ -44,9 +55,38 @@ public void should_load_pacts_from_test() {
                 .hasFieldOrPropertyWithValue("consumer", new Consumer("planets_consumer"));
     }
 
+    @Test
+    public void should_load_pacts_from_algeron_provider_configuration() {
+        final PactsReader pactsReader = new PactsReader();
+        pactsReader.pactsInstanceProducer = instanceProducer;
+        pactsReader.algeronProviderConfigurationInstance = instance;
+
+
+        String retriever = "provider: folder" + System.lineSeparator() + "contractsFolder: pacts";
+        final Map<String, String> configuration = new HashMap<>();
+        configuration.put("retrieverConfiguration", retriever);
+
+        final AlgeronProviderConfiguration algeronProviderConfiguration = AlgeronProviderConfiguration.fromMap(configuration);
+
+        when(instance.get()).thenReturn(algeronProviderConfiguration);
+
+        pactsReader.readPacts(new BeforeClass(NonePactDefinition.class));
+        verify(instanceProducer).set(argumentCaptor.capture());
+
+        Pacts pacts = argumentCaptor.getValue();
+        final List<Pact> listOfLoadedPacts = pacts.getPacts();
+        assertThat(listOfLoadedPacts).hasSize(1).element(0)
+                .hasFieldOrPropertyWithValue("provider", new au.com.dius.pact.model.Provider("planets_provider"))
+                .hasFieldOrPropertyWithValue("consumer", new Consumer("planets_consumer"));
+    }
```

Comment:
Just a general suggestion - if you group code in some "logical" blocks which I believe are representing certain groups of operations maybe we can add ``given/when/then` comments too? At least for me it would be easier to scan it :)